services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -1,18 +1,37 @@
 package cromwell.pipeline.service
 
 import cromwell.pipeline.datastorage.dao.repository.DocumentRepository
+import cromwell.pipeline.datastorage.dto.project.configuration.{
+  ProjectConfigurationRequest,
+  ProjectConfigurationResponse
+}
 import cromwell.pipeline.datastorage.dto.{ ProjectConfiguration, ProjectId }
 
 import scala.concurrent.{ ExecutionContext, Future }
 
 class ProjectConfigurationService(repository: DocumentRepository)(implicit ec: ExecutionContext) {
   import ProjectConfiguration._
 
-  def addConfiguration(projectConfiguration: ProjectConfiguration): Future[String] =
+  def addConfiguration(configuration: ProjectConfigurationRequest): Future[String] =
     repository
-      .updateOne(toDocument(projectConfiguration), "projectId", projectConfiguration.projectId.value)
+      .replaceOne(
+        toDocument(ProjectConfiguration(configuration.projectId, configuration.projectFileConfigurations)),
```

Comment:
It's better to have an apply method in ProjectConfiguration companion object which takes `ProjectConfigurationResponse`.